src/composition/supergraph/config/resolver/mod.rs

```diff
@@ -117,7 +120,59 @@ impl SupergraphConfigResolver<state::LoadRemoteSubgraphs> {
                 .call(FetchRemoteSubgraphsRequest::new(graph_ref.clone()))
                 .await
                 .map_err(|err| {
-                    LoadRemoteSubgraphsError::FetchRemoteSubgraphsError(Box::new(err))
+
+                    // Q1: Academic: Why choose map_err here instead of From? I think I could use
+                    // the From trait to cast the error raised by the try operator (?) into
+                    // whatever I want. I'm chosing to use map_err here becuase it is how the rest
+                    // of this code is structure, but curious about the idiomatic approach here.
+
+
+                    // Q2: Practical: I want to write a match statement that has an arm that matches
+                    // the `err` has an "Invalid credentials provided" message. I've copied what
+                    // `err` logs to the console below.
+
+                    // eprintln!("Error => \n {:?}", err);
+                    // Error =>
+                    // Service {
+                    //  source: PartialError {
+                    //      data: ResponseData {
+                    //          variant: None
+                    //      },
+                    //      errors: [
+                    //          Error {
+                    //              message: "406: Not Acceptable",
+                    //              locations: None,
+                    //              path: None,
+                    //              extensions:
+                    //                  Some({
+                    //                      "response": Object {
+                    //                          "body": Object {
+                    //                              "errors": Array [
+                    //                                  Object {
+                    //                                      "message": String("Invalid credentials provided")
+                    //                                  }
+                    //                               ]
+                    //                           },
+                    //                           "status": Number(406),
+                    //                           "statusText": String("Not Acceptable"),
+                    //                           "url": String("http://engine-graphql.apollo-default.svc.prod.apollo-internal:8081/api/graphql")},
+                    //                           "code": String("INTERNAL_SERVER_ERROR")
+                    //                  })
+                    //          }
+                    //      ]
+                    //  },
+                    //  endpoint_kind: ApolloStudio
+                    // }
+
+                    // psuedo: what I'm trying to accomplish
+
+                    //match err {
+                        //PartialError?!?(_) => { LoadRemoteSubgraphsError::FetchRemoteSubgraphsAuthError(Box::new(err)) },
```

Comment:
I want this first arm to match when `source.errors[*]...errors.message == "Invalid credentials provided"`

OR

I want the idiomatic way to raise `LoadRemoteSubgraphsError::FetchRemoteSubgraphsAuthError(Box::new(err))` when the err includes the string "Invalid credentials provided", otherwise return `LoadRemoteSubgraphsError::FetchRemoteSubgraphsError(Box::new(err))`